components/locid/src/extensions/mod.rs

```diff
@@ -74,7 +80,7 @@ impl ExtensionType {
             b'u' => Ok(Self::Unicode),
             b't' => Ok(Self::Transform),
             b'x' => Ok(Self::Private),
-            _ => Err(ParserError::InvalidExtension),
+            _ => Ok(Self::Other(key)),
         }
```

Comment:
`suggestion`: this becomes infallible.

Reply:
I need to check that the key is 'a' .. 'z', and throw an error otherwise here.